backend/src/org/commcare/suite/model/Detail.java

```diff
@@ -64,6 +65,22 @@ public Detail(
             String id, DisplayUnit title,
             Vector<Detail> details,
             Vector<DetailField> fields,
+            OrderedHashtable<String, String> variables, Action action, Callout callout
+    ) {
+        this(
+                id, title,
+                ArrayUtilities.copyIntoArray(details, new Detail[details.size()]),
+                ArrayUtilities.copyIntoArray(fields, new DetailField[fields.size()]),
```

Comment:
Why the need to call copyIntoArray? Can't the vectors be passed directly into the constructor defined below this one?